File: Array/2570. Merge Two 2D Arrays by Summing Values.cpp
class Solution {
    public:
        vector<vector<int>> mergeArrays(vector<vector<int>>& a, vector<vector<int>>& b) 
        {
            map<int,int>mp;
            for(auto x:a) mp[x[0]] += x[1];
            for(auto x:b) mp[x[0]] += x[1];
    
            vector<vector<int>> ans;
            for(auto x:mp) ans.push_back({x.first,x.second});
            return ans;
        }
    };

File: 2570. Merge Two 2D Arrays by Summing Values.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        store = defaultdict(int)
        for id, val in nums1:
            store[id] += val
        for id, val in nums2:
            store[id] += val

        ans = []
        for id, val in store.items():
            ans.append([id, val])
        ans.sort()
        return ans